src/sentry/api/endpoints/organization_events_facets_performance.py

```diff
@@ -128,13 +128,30 @@ def get(self, request, organization):
             return Response([])
 
         tag_key = request.GET.get("tagKey")
+        tag_key_limit = request.GET.get("tagKeyLimit")
+        num_buckets_per_key = request.GET.get("numBucketsPerKey")
+
+        if not tag_key_limit:
+            raise ParseError(detail="'tagKeyLimit' must be provided for the performance histogram.")
+        if not num_buckets_per_key:
+            raise ParseError(
+                detail="'numBucketsPerKey' must be provided for the performance histogram."
+            )
+
+        tag_key_limit = int(tag_key_limit)
+        num_buckets_per_key = int(num_buckets_per_key)
+
+        if tag_key_limit * num_buckets_per_key > 500:
+            raise ParseError(
+                detail="The number of total buckets ('tagKeyLimit' * 'numBucketsPerKey') cannot exceed 500"
+            )
 
         if not tag_key:
             raise ParseError(detail="'tagKey' must be provided when using histograms.")
 
-        def data_fn(offset, limit):
+        def data_fn():
```

Comment:
Yeah, good question. I left it because I'm not entirely sure. I think we might since after playing with it with a lot of data I'm not sure we can get away without showing the rest. More testing required.